ConfigurationSubstitution.Tests/ConfigurationTests.cs

```diff
@@ -370,5 +370,187 @@ public void Should_get_substituted_value_when_using_long_substituable_pattern(Fu
 
             substituted.Should().Be("Hello world");
         }
+
+        [Theory]
+        [MemberData(nameof(ConfigurationBuilderTheoryData))]
+        public void Should_substitute_when_delimited_fallback_default_value_provided(Func<IConfigurationBuilder> builderGenerator)
+        {
+            var configurationBuilder = builderGenerator()
+                .AddInMemoryCollection(new Dictionary<string, string>()
+                {
+                    { "Foo", "Hello %(env:Testo)%" },
+                })
+                .EnableSubstitutionsWithDelimitedFallbackDefaults("%(", ")%", ":");
+
+            var configuration = configurationBuilder.Build();
+
+            // Act
+            var substituted = configuration["Foo"];
+
+            substituted.Should().Be("Hello Testo");
+        }
+
+        [Theory]
+        [MemberData(nameof(ConfigurationBuilderTheoryData))]
+        public void Should_substitute_variable_when_provided_and_not_fallback_default_value(Func<IConfigurationBuilder> builderGenerator)
+        {
+            var configurationBuilder = builderGenerator()
+                .AddInMemoryCollection(new Dictionary<string, string>()
+                {
+                    { "Foo", "Hello $(Var1:Testo)" },
+                    { "Var1", "world" }
+                })
+                .EnableSubstitutionsWithDelimitedFallbackDefaults("$(", ")", ":");
+
+            var configuration = configurationBuilder.Build();
+
+            // Act
+            var substituted = configuration["Foo"];
+
+            substituted.Should().Be("Hello world");
+        }
+
+        [Theory]
+        [MemberData(nameof(ConfigurationBuilderTheoryData))]
+        public void Should_substitute_when_fallback_default_value_is_empty(Func<IConfigurationBuilder> builderGenerator)
+        {
+            var configurationBuilder = builderGenerator()
+                .AddInMemoryCollection(new Dictionary<string, string>()
+                {
+                    { "Foo", "$(Var1:)" },
+                })
+                .EnableSubstitutionsWithDelimitedFallbackDefaults("$(", ")", ":");
+
+            var configuration = configurationBuilder.Build();
+
+            // Act
+            var substituted = configuration["Foo"];
+
+            substituted.Should().Be(string.Empty);
+
+        }
+
+        [Theory]
+        [MemberData(nameof(ConfigurationBuilderTheoryData))]
+        public void Should_throw_exception_when_fallback_default_value_delimiter_is_null(Func<IConfigurationBuilder> builderGenerator)
+        {
+            var configurationBuilder = builderGenerator()
+                .AddInMemoryCollection(new Dictionary<string, string>()
+                {
+                    { "Foo", "$(Var1:Testo)" },
+                })
+                .EnableSubstitutionsWithDelimitedFallbackDefaults("$(", ")", null);

```

Comment:
Question: wouldn't it make more sense that in both `.EnableSubstitutions` and `.EnableSubstitutionsWithDelimitedFallbackDefaults` we always throw if any argument is null or empty?

Reply:
Good point. Yeah, I think that makes perfect sense. This highlighted test only looks at the `fallbackDefaultValueDelimiter` argument, but I think think the remaining two arguments, namely `substitutableStartsWith` and `substitutableEndsWith`, should be included too and throw an exception when they're empty or null.

Reply:
If you could do that directly in the extension method then, failing fast is good
PR quality is really good, thanks a lot!

Reply:
Cool! Will do and commit that asap
Thank you very much for taking the time to review and build this awesome project, I'm glad I could contribute to it 🙂

Reply:
Just committed  e3b7ed9e0ca8050ff7498fb2203aa0062b7c4b58, which has the discussed suggestions along some unit tests to verify the correct behaviour. Please take a look and see if anything is missing.